Common/NearestNeighbour/src/main/java/org/tribuo/common/nearest/KNNTrainer.java

```diff
@@ -85,14 +105,35 @@ private KNNTrainer() {}
     /**
      * Creates a K-NN trainer using the supplied parameters.
      * @param k The number of nearest neighbours to consider.
+     * @param distType The distance function.
+     * @param numThreads The number of threads to use.
+     * @param combiner The combination function to aggregate the k predictions.
+     * @param backend The computational backend.
+     */
+    public KNNTrainer(int k, DistanceType distType, int numThreads, EnsembleCombiner<T> combiner, Backend backend) {
+        this.k = k;
+        this.distType = distType;
+        this.numThreads = numThreads;
+        this.combiner = combiner;
+        this.backend = backend;
+        postConfig();
+    }
+
+    /**
+     * Creates a K-NN trainer using the supplied parameters.
+     * @deprecated
+     * This Constructor is deprecated in version 4.3.
+     *
+     * @param k The number of nearest neighbours to consider.
      * @param distance The distance function.
      * @param numThreads The number of threads to use.
      * @param combiner The combination function to aggregate the k predictions.
      * @param backend The computational backend.
      */
+    @Deprecated
     public KNNTrainer(int k, Distance distance, int numThreads, EnsembleCombiner<T> combiner, Backend backend) {
         this.k = k;
-        this.distance = distance;
+        this.distType = distance.getDistanceType();
```

Comment:
Maybe this one and the similar one in `KMeansTrainer` should call `this(k, distance.getDistanceType(), numThreads, combiner, backend)`? That'll remove some redundant code.